homeassistant/components/coinbase.py

```diff
@@ -0,0 +1,83 @@
+"""
+Support for Coinbase.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/coinbase/
+"""
+from datetime import timedelta
+
+import logging
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_API_KEY
+from homeassistant.util import Throttle
+from homeassistant.helpers.discovery import load_platform
+
+REQUIREMENTS = ['coinbase==2.0.6']
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'coinbase'
+
+CONF_API_SECRET = 'api_secret'
+CONF_EXCHANGE_CURRENCIES = 'exchange_rate_currencies'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
+
+DATA_COINBASE = 'coinbase_cache'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Required(CONF_API_SECRET): cv.string,
+        vol.Optional(CONF_EXCHANGE_CURRENCIES, default=[]):
+            vol.All(cv.ensure_list, [cv.string])
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the Coinbase component.
+
+    Will automatically setup sensors to support
+    wallets discovered on the network.
+    """
+    api_key = config[DOMAIN].get(CONF_API_KEY)
+    api_secret = config[DOMAIN].get(CONF_API_SECRET)
+    exchange_currencies = config[DOMAIN].get(CONF_EXCHANGE_CURRENCIES)
+
+    hass.data[DATA_COINBASE] = coinbase_data = CoinbaseData(api_key,
```

Comment:
What happens here if the credentials are bad and the auth process fails? Can we determine that and return `False` here from component setup?